sovtokenfees/sovtokenfees/test/catchup/revert/test_revert_fees.py

```diff
@@ -0,0 +1,84 @@
+from threading import Thread
+
+import pytest
+from sovtoken.constants import XFER_PUBLIC, ADDRESS, AMOUNT, SEQNO, OUTPUTS
+from sovtokenfees.constants import FEES
+from sovtokenfees.test.helper import check_state, ensure_all_nodes_have_same_data, send_and_check_nym_with_fees
+
+from indy_common.constants import NYM
+from plenum.common.startable import Mode
+from plenum.common.txn_util import get_seq_no
+from plenum.test.delayers import cDelay
+from plenum.test import waits
+from plenum.test.helper import assertExp
+from plenum.test.stasher import delay_rules
+from plenum.test.test_node import ensureElectionsDone
+from plenum.test.view_change.helper import ensure_view_change
+from stp_core.loop.eventually import eventually
+
+
+def test_revert_fees_with_xfer(nodeSetWithIntegratedTokenPlugin, xfer_mint_tokens,
+                               fees_set, helpers, looper, xfer_addresses):
+    nodes = nodeSetWithIntegratedTokenPlugin
+    node_set = [n.nodeIbStasher for n in nodeSetWithIntegratedTokenPlugin]
+
+    addr = xfer_addresses[0]
+    seq_no = get_seq_no(xfer_mint_tokens)
+    utxo = [{
+        ADDRESS: addr,
+        SEQNO: seq_no
+    }]
+
+    outputs = [{
+        ADDRESS: xfer_addresses[1],
+        AMOUNT: 998
+    }]
+
+    with delay_rules(node_set, cDelay()):
+        # should be changed for auth rule
+        helpers.general.set_fees_without_waiting({XFER_PUBLIC: 2})
+        looper.runFor(waits.expectedPrePrepareTime(len(node_set)))
```

Comment:
In order to have deterministic test, please wait for PrePrepare being applied instead.
You can use, for example, 
`looper.run(eventually(check_prepare_certificate, node_set, initial_pp_seq_no+1))`